internal/infrastructure/kubernetes/ratelimit/resource_provider.go

```diff
@@ -186,19 +187,28 @@ func (r *ResourceRender) ServiceAccount() (*corev1.ServiceAccount, error) {
 // Deployment returns the expected rate limit Deployment based on the provided infra.
 func (r *ResourceRender) Deployment() (*appsv1.Deployment, error) {
 	containers := expectedRateLimitContainers(r.rateLimit, r.rateLimitDeployment, r.Namespace)
-	labels := rateLimitLabels()
-	selector := resource.GetSelector(labels)
+	selector := resource.GetSelector(rateLimitLabels())
+
+	podLabels := rateLimitLabels()
+	if r.rateLimitDeployment.Pod.Labels != nil {
+		maps.Copy(podLabels, r.rateLimitDeployment.Pod.Labels)
+		maps.Copy(podLabels, rateLimitLabels())
```

Comment:
can L195 be removed ? since L192 is already setting it

Reply:
So `maps.Copy(dest, src)` [overwrites keys existing ](https://pkg.go.dev/maps#Copy) in both maps with the `src` value. I want to make sure noone can override one of the selector labels from the `rateLimitLabels`, so I'm doing the last maps.Copy to overwrite with the selector labels in case someone added one of those labels in their values.

Also the destination map must not be nil so I cannot do something like:
```
podLabels := r.rateLimitDeployment.Pod.Labels
maps.Copy(podLabels, rateLimitLabels())
```
as it will fail if no pod labels are provided.
It looks a bit funny, I just didn't want to implement a function that merges two maps, but maybe it's worth it for clarity's sake. Let me know if you think so.

Reply:
ah thats a good way to ensure the labels can't be overridden
nit: this as a code comment would be helpful so the next dev gets the "why"